lib/topic_query.rb

```diff
@@ -338,6 +338,14 @@ def list_bookmarks
     create_list(:bookmarks) { |l| l.where("tu.bookmarked") }
   end
 
+  def list_hot
+    create_list(:hot, unordered: true) do |topics|
```

Comment:
I think this list should have a maximum number of results. Not sure exactly what, but 200-300 would be enough IMO.